unittests/test_pipeline.py

```diff
@@ -287,6 +287,23 @@ def test_supports_system(self):
         assert not test.supports_system('testsys:gpu')
         assert not test.supports_system('testsys:login')
 
+        test.valid_systems = ['*:gpu']
+        assert test.supports_system('testsys:gpu')
+        assert test.supports_system('foo:gpu')
+        assert not test.supports_system('testsys:cpu')
+        assert not test.supports_system('testsys:login')
+
+        test.valid_systems = ['*:*']
+        assert test.supports_system('gpu')
+        assert test.supports_system('login')
+        assert test.supports_system('testsys:gpu')
+        assert test.supports_system('testsys:login')
+
+        test.valid_systems = ['testsys:*']
+        assert test.supports_system('testsys:login')
+        assert test.supports_system('gpu')
+        assert not test.supports_system('foo:gpu')
+
```

Comment:
Can you also add a test for testing a system that is of the form `x:y:z`?

Reply:
What should be the result of this test ?

Reply:
This is entirely up to us, but it should definitely not crash. I would suggest `testsys:login:foo` to compare to false, which is what the current implementation would do in any case if you use `maxsplit=1`.

Reply:
Mmm, I would argue that the logical result should be that it does give an error, not a false result

Reply:
It's not a false result. We only consider a single `:` separator and whatever is left to it is the system name, whatever is right to it is the partition name, then it's consistent if in the case of `testsys:login:foo` you assume that `testsys` is the system name and `login:foo` is the partition name. So the test should compare false.

Reply:
So, `login:foo` would be a valid partition name ?

Reply:
Although I'm not entirely convinced of this logic, since we are really talking about a corner case, I figured out that we can quite easily validate the in `valid_systems`, which I think is the best solution and we will be both happy. If you change the `str` [here](https://github.com/eth-cscs/reframe/blob/master/reframe/core/pipeline.py#L165) to `typ.Str[r'proper_regex']`, we're good to go. If `valid_systems` are not correct, you'll get an error right from the beginning. So you can skip this unit test completely.

Reply:
This regex seems to do the job: `^[^:]+(:[^:]+)?$`. I don't know if we could write in a simpler way.

Reply:
I've just done the change and pushed.

Reply:
I checked the code a bit more in detail. the `valid_systems` definitions does not have to change. The argument to the `supports_system` comes essentially from the configuration file through the `SystemPartition.fullname` property and there is no way it can be wrongly formatted due to the user. If it is, then it's a framework bug. So I will revert my change above and will approve the PR as is.